### DISCUSSION TITLE
dispatchSync dispatches jobs on `sync` queue.

### DISCUSSION THREAD

**Discussion by nick-nds** [laravel/framework#51128] [Q&A]
I encountered an issue while writing a test for a Job. I attempted to dispatch the job using the `Job::dispatchSync()` function, but the `handle` method of the Job was not being executed. Upon investigation, I discovered that `dispatchSync` dispatches the job on a sync queue. While I'm unsure of the reasoning behind this implementation, I need a way to manually dispatch the job from unit tests.

Unfortunately, I cannot use `dispatchNow` on the Job as it is not available in the `Dispatchable` trait. While I could manually call the `handle` function on the Job class, I would prefer to avoid this approach.

Are there any alternative methods for testing the logic of Job classes?

Edit: I'm seeking to understand why jobs dispatched with dispatchSync are placed on a sync queue instead of using dispatchNow.

**Comment by y8a5y** [comment]
I don't see why you want to avoid calling `Job::handle()` directly, that's basically what `dispatchNow` was doing anyway. In case you didn't know, `dispatchNow` was deprecated long ago and has already been removed from current version of Laravel.

One drawback of the old `dispatchNow` method is that it wouldn't trigger job's "events" (e.g. job middlewares, `failed` method), I think that's one of the reasons `dispatchSync` now uses a queue, jobs are actually treated like jobs and not just like a fancy callable.

**Comment by tasmanboat** [comment]
Consider use Bus to fire dispatchNow. The dispatchNow() method executes the job synchronously without interacting with any queues. ref [1](https://stackoverflow.com/questions/59578987/laravel-job-queue-dispatch-now-not-failing-like-normal) [2](https://laracasts.com/discuss/channels/laravel/optionally-process-job-synchronously)
```
use Illuminate\Support\Facades\Bus;

Bus::dispatchNow(new Job($param));
```

Or just call your job like a normal class. ref [1](https://github.com/laravel/framework/issues/37023#issuecomment-822453031)
```
$result = (new TestJob())->handle();
```
